src/streams.h

```diff
@@ -138,6 +138,80 @@ class CVectorWriter
     size_t nPos;
 };
 
+/* Minimal stream for reading from an existing vector by reference
+ */
+class CVectorReader
```

Comment:
In commit "streams: Create CVectorReader stream interface for vectors." (93f702b08e413c5c025b155bfb62b721d27939f5)

This is pretty similar to the VectorReader class @TheBlueMatt is adding here: https://github.com/TheBlueMatt/bitcoin/commit/bb608a995e8fd16d156145bf41023e7a77d44971 for https://github.com/bitcoin/bitcoin/compare/master...TheBlueMatt:2018-02-miningserver

Your implementation is more general with support for deserialization in the constructor and more complete comments. But his has a `pos()` method and uses non-hungarian names which are recommended by the contrib guide. Anyway you may want to incorporate his changes.

Reply:
OK, I can bring that commit over instead or modify this one to remove the hungarian notation.

Reply:
Done.